handlers/root.go

```diff
@@ -5,9 +5,110 @@ import (
 	"net/http"
 
 	"github.com/swift-sunshine/swscore/log"
+	"github.com/swift-sunshine/swscore/pkg/kubernetes"
+	"github.com/swift-sunshine/swscore/pkg/prometheus"
+
 )
 
+// This method shows how to use the API to fetch and print all info that would be necessary
+// for listing services and details.
+// It is temporal, so don't be afraid to delete it when handlers are in place.
+func demoFetchAllServicesAndPrintThem(w http.ResponseWriter) {
+	istioClient, err := kubernetes.NewClient()
+	if err != nil {
+		log.Error(err)
+		return
+	}
+	prometheusClient, err := prometheus.NewClient()
+	if err != nil {
+		log.Error(err)
+		return
+	}
+
+	namespaces, err := istioClient.GetNamespaces()
+	if err != nil {
+		log.Error(err)
+		return
+	}
+	fmt.Fprintf(w, "Namespaces: %v \n\n", namespaces)
+
+	for _, namespace := range namespaces {
+		services, err := istioClient.GetServices(namespace)
+		if err != nil {
+			log.Error(err)
+			return
+		}
+		fmt.Fprintf(w, "Namespace: %s Services %v \n\n", namespace, services)
+
+		for _, service := range services {
+			fmt.Fprintf(w,"Service Name: %s \n", service)
+			details, err := istioClient.GetServiceDetails(namespace, service)
+			if err != nil {
+				log.Error(err)
+				return
+			}
+			fmt.Fprintf(w,"Service Labels: \n")
+			for key, value := range details.Service.Labels {
+				fmt.Fprintf(w,"%s = %s \n", key, value)
+			}
+			fmt.Fprintf(w,"Type: %s \n", details.Service.Spec.Type)
+			fmt.Fprintf(w,"IP: %s \n", details.Service.Spec.ClusterIP)
+
+			fmt.Fprintf(w,"Ports: \n")
+			for _, port := range details.Service.Spec.Ports {
+				fmt.Fprintf(w,"%s %d %s \n", port.Protocol, port.Port, port.Name)
+			}
+
+			fmt.Fprintf(w,"Endpoints: \n")
+
+			for _, subset := range details.Endpoints.Subsets {
+				for _, address := range subset.Addresses {
+					fmt.Fprintf(w,"Address: %s \n", address.IP)
+					if address.TargetRef != nil {
+						fmt.Fprintf(w,"Kind: %s \n", address.TargetRef.Kind)
+						fmt.Fprintf(w,"Name: %s \n", address.TargetRef.Name)
+					}
+				}
+				for _, port := range subset.Ports {
+					fmt.Fprintf(w,"Port: %d \n", port.Port)
+					fmt.Fprintf(w,"Protocol: %s \n", port.Protocol)
+					fmt.Fprintf(w,"Name: %s \n", port.Name)
+				}
+			}
+
+			fmt.Fprintf(w,"Pods: \n")
+
+			for _, pod := range details.Pods {
+				fmt.Fprintf(w,"Pod: %s \n", pod.Name)
+				fmt.Fprintf(w,"Pod Labels: \n")
+				for key, value := range pod.Labels {
+					fmt.Fprintf(w,"%s = %s \n", key, value)
+				}
+			}
+
+			fmt.Fprintf(w,"Istio Rules: \n")
+
+			istioDetails, err := istioClient.GetIstioDetails(namespace, service)
+			if err != nil {
+				log.Error(err)
+				return
+			}
+			for _, rule := range istioDetails.RouteRules {
+				fmt.Fprintf(w,"RouteRule: %+v \n", rule.Spec)
+			}
+
+			fmt.Fprintf(w,"Dependencies: \n")
+			incomeServices, err := prometheusClient.GetIncomeServices(namespace, service)
```

Comment:
not checking err

note also - using the if x, err := condition; err != nil {} else {} is a typical go construct (rather than putting the if err!=nil check on a separate line - this has the added bonus of  putting the err definition within the scope of the if-statement rather than putting it in the outer block where the if stmt is located - most times you don't want the err leaking out to the outer block).

Reply:
+1 
Thanks, I'll update the PR with that pattern, I haven't learnt yet the most common ones.

Reply:
> note also - using the if x, err := condition; err != nil {} else {} is a typical go construct

I've reviewed this proposed style and in some places makes sense but in other perhaps it can increase the nested conditional sentences, and make more hard to read.
I.e. if you need x several places, instead to nest the call, you can return on error.

So I will change when I think it's clear.